### ISSUE TITLE
London and Paris Marathon are quite heavy (25MB)

### ISSUE THREAD

**Issue by ThomasParistech** [ThomasParistech/pretty-gpx#59]
When the map is really large and centered on a dense city (like Paris or London Marathon), we could avoid drawing each tiny street, to lighten the filesize.

What do you think @JulesL2  ?

**Comment by JulesL2** [comment]
Indeed, I will check the options to see how we can detect it is a city (probably by doing a count request in overpass) and how we can fix it :)

**Comment by JulesL2** [comment]
@ThomasParistech Do you speak of the download data size ?
Or only the vectorized image size ?

If it is the vectorized image size then I can also investigate a filtering of points/ a resampling when drawing roads.

**Comment by ThomasParistech** [comment]
I meant the vectorized image size
When I open it on my laptop, the viewer struggles a bit ^^

**Comment by JulesL2** [comment]
I have included both.

One thing : To chose the priority level to donwload I have a function (such as an auto mode) it would be cool to be able to override the auto value with a user value in the gui if someone really want a high res pic.
Second thing : The criteria in the auto mode should be tested on a lot of maps, I have tested only 4-5 maps.
Third thing : Seems that the download is still the limiting factor but simplifying the lines add a bit of complexicity

**Comment by JulesL2** [comment]
It seems that a lot of ways are not merged, I will have a look

**Comment by ThomasParistech** [comment]
i was using Chrome

**Comment by JulesL2** [comment]
@ThomasParistech Do you see any improvements with PR : https://github.com/ThomasParistech/pretty-gpx/pull/76

**Comment by ThomasParistech** [comment]
@JulesL2 
It's indeed a bit smoother
But Chrome seems to have trouble drawing so many very tiny lines

It looks like a tough anti-aliasing problem. We try to draw too high frequencies given the printing resolution.
If the Paris Marathon poster is printed in A4, then it means the small cutout below, with rich details will be printed on a 1cmx1cm square.


What do you think ?

![image](https://github.com/user-attachments/assets/c3cea7c0-d1a8-40ed-b4e0-7ec6b703440c)

![image](https://github.com/user-attachments/assets/351de4f1-3d1d-4e83-850b-b8ada5bfccee)

**Comment by JulesL2** [comment]
@ThomasParistech I did the work, I let you check the PR https://github.com/ThomasParistech/pretty-gpx/pull/91 , there is still a bit of work to do to make things better coded but the feature is implemented

**Comment by JulesL2** [comment]
@ThomasParistech The feature is ready to be merged, I let you check the gain you have in poster readability in your viewer. On my side we can obtain a good reduction of the poster size.

**Comment by ThomasParistech** [comment]
@JulesL2 
Will look at the PRs Wednesday
Thanks!
